### PR TITLE
Avkr003/perp fixes

### PR DESCRIPTION
**Pull request by avkr003**

### CODE REVIEW

File: x/accountedpool/keeper/hooks_leveragelp.go [hunk 1]
```diff
@@ -76,14 +76,14 @@ func (h LeverageLpHooks) AfterDisablingPool(ctx sdk.Context, ammPool ammtypes.Po
 	return h.k.OnLeverageLpPoolDisable(ctx, ammPool)
 }
 
-func (h LeverageLpHooks) AfterLeverageLpPositionOpen(ctx sdk.Context, sender sdk.AccAddress) error {
-	return nil
+func (h LeverageLpHooks) AfterLeverageLpPositionOpen(ctx sdk.Context, sender sdk.AccAddress, ammPool ammtypes.Pool) error {
+	return h.k.UpdateAccountedPoolOnAmmChange(ctx, ammPool)
```
[the remark this answers is not in the document]
Comment (avkr003) [reply x/accountedpool/keeper/hooks_leveragelp.go:80]: You are right, we are.

File: x/accountedpool/keeper/hooks_leveragelp.go [hunk 2]
```diff
@@ -76,14 +76,14 @@ func (h LeverageLpHooks) AfterDisablingPool(ctx sdk.Context, ammPool ammtypes.Po
 	return h.k.OnLeverageLpPoolDisable(ctx, ammPool)
 }
 
-func (h LeverageLpHooks) AfterLeverageLpPositionOpen(ctx sdk.Context, sender sdk.AccAddress) error {
+func (h LeverageLpHooks) AfterLeverageLpPositionOpen(ctx sdk.Context, sender sdk.AccAddress, ammPool ammtypes.Pool) error {
```
Comment (amityadav0) [inline x/accountedpool/keeper/hooks_leveragelp.go:79]:
```suggestion
func (h LeverageLpHooks) AfterLeverageLpPositionOpen(ctx sdk.Context, sender sdk.AccAddress) error {
```

Comment (amityadav0) [reply x/accountedpool/keeper/hooks_leveragelp.go:79]: this is not required, right ?

File: x/accountedpool/keeper/hooks_leveragelp.go [hunk 3]
```diff
@@ -76,14 +76,14 @@ func (h LeverageLpHooks) AfterDisablingPool(ctx sdk.Context, ammPool ammtypes.Po
...
 	return nil
 }
 
-func (h LeverageLpHooks) AfterLeverageLpPositionClose(ctx sdk.Context, sender sdk.AccAddress) error {
+func (h LeverageLpHooks) AfterLeverageLpPositionClose(ctx sdk.Context, sender sdk.AccAddress, ammPool ammtypes.Pool) error {
```
Comment (amityadav0) [inline x/accountedpool/keeper/hooks_leveragelp.go:83]: same

File: x/accountedpool/keeper/hooks_leveragelp.go [hunk 4]
```diff
@@ -76,14 +76,14 @@ func (h LeverageLpHooks) AfterDisablingPool(ctx sdk.Context, ammPool ammtypes.Po
...
 	return nil
 }
 
-func (h LeverageLpHooks) AfterLeverageLpPositionOpenConsolidate(ctx sdk.Context, sender sdk.AccAddress) error {
+func (h LeverageLpHooks) AfterLeverageLpPositionOpenConsolidate(ctx sdk.Context, sender sdk.AccAddress, ammPool ammtypes.Pool) error {
```
Comment (amityadav0) [inline x/accountedpool/keeper/hooks_leveragelp.go:87]: same

File: x/leveragelp/keeper/msg_server_close.go [hunk 5]
```diff
@@ -21,7 +21,11 @@ func (k Keeper) Close(ctx sdk.Context, msg *types.MsgClose) (*types.MsgCloseResp
 	}
 
 	if k.hooks != nil {
-		err := k.hooks.AfterLeverageLpPositionClose(ctx, sdk.MustAccAddressFromBech32(msg.Creator))
+		ammPool, err := k.GetAmmPool(ctx, closedPosition.AmmPoolId)
+		if err != nil {
+			return nil, err
+		}
```
Comment (amityadav0) [inline x/leveragelp/keeper/msg_server_close.go:27]: this part can also be removed

Comment (amityadav0) [reply x/leveragelp/keeper/msg_server_close.go:27]: below too

**Review by amityadav0 — CHANGES_REQUESTED** [review]

**Comment by Wgil** [comment]
Should we include https://elys.atlassian.net/browse/DEV-2106 migration as part of this PR?

File: x/accountedpool/keeper/hooks_leveragelp.go (hunk 3 above)
Comment (avkr003) [reply x/accountedpool/keeper/hooks_leveragelp.go:83]: same

File: x/accountedpool/keeper/hooks_leveragelp.go (hunk 4 above)
Comment (avkr003) [reply x/accountedpool/keeper/hooks_leveragelp.go:87]: same

**Comment by avkr003** [comment]
@Wgil It might be slightly tricky and risky to implement for the upcoming upgrade. If we miss something important, it can mess up other things too. I am adding a temporary solution to query all pools

**Pull request merged**